### DISCUSSION TITLE
Setter of bound property is not called if value has not changed

### DISCUSSION THREAD

**Discussion by d-mackenzie** [AvaloniaUI/Avalonia#16603] [Q&A] [answered]
XAML:
```
<NumericUpDown Value="{Binding Number}" Minimum="1" Maximum="200" FormatString="N0" ShowButtonSpinner="False"/>
```

ViewModel C#:
```
public int Number
{
	get { return _number; }
	set { _number = value; RaisePropertyChanged(); }
}

// snip...

protected void RaisePropertyChanged([CallerMemberName] string propertyName = "")
{
    if (PropertyChanged != null)
        PropertyChanged(this, new PropertyChangedEventArgs(propertyName));
}
```

When the form loads:
![image](https://github.com/user-attachments/assets/120ad178-784c-486f-9c08-1cd066e84f0a)

Press BACKSPACE to clear the NumericUpDown control:
![image](https://github.com/user-attachments/assets/e64b0186-3025-4a1e-84ad-0051acee69c0)

Type 1 into the NumericUpDown control:
![image](https://github.com/user-attachments/assets/ccdebf73-8879-44a6-b8e3-976693d45663)

As you can see, the error remains. I think this is because the property in the ViewModel is still 1 -- so it hasn't changed. The setter in the ViewModel isn't even called. The same happens whenever you set the text in the control to what is was before it was deleted.

In my project, things are a bit more complex: I have a TextBox with a ValueConverter and a DataAnnotation on the property to enforce the range and customize the error message. However, the same behaviour is seen with just a NumericUpDown and simple backing field.

Is there a way to force the setter to be called or otherwise remove the error when the control is in a valid state? Maybe a property of the control, or an annotation on the ViewModel property, or similar?

Any suggestions appreciated!

Thanks,
David.

**Comment by maxkatz6** [comment, marked as answer]
Looks like this issue https://github.com/AvaloniaUI/Avalonia/issues/16229

**Reply by d-mackenzie** [reply]
Thanks for the swift reply, Max.

That issue exactly describes the problem. It's not a massive problem for my project, happy to wait until the fix finds its way into a release.

David.
